src/main/kotlin/net/lumalyte/lg/infrastructure/listeners/RoseChatCleanupListener.kt

```
@@ -0,0 +1,69 @@
+            // Check if the channel is guild-related or ally-related
+            // Matches 'guild_', 'ally_', or 'officer_' patterns used by Luma/RoseChat hooks
+            val channelId = currentChannel.id.lowercase()
+            if (channelId.contains("guild") || channelId.contains("ally") || channelId.contains("officer")) {
+                val defaultChannel = api.channelManager.defaultChannel
+                if (defaultChannel != null && currentChannel.id != defaultChannel.id) {
+                    rosePlayer.switchChannel(defaultChannel)
+                }
+            }
+        } catch (e: NoClassDefFoundError) {
+            // RoseChat not loaded
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)